consensus/istanbul/backend/announce.go

```diff
@@ -302,83 +390,92 @@ func (sb *Backend) handleIstAnnounce(payload []byte) error {
 	}
 	logger.Trace("Handling an IstanbulAnnounce message", "from", msg.Address)
 
-	// If the message is originally from this node, then ignore it
-	if msg.Address == sb.Address() {
-		logger.Trace("Received an IstanbulAnnounce message originating from this node. Ignoring it.")
-		return nil
-	}
-
-	var announceData announceData
-	err = rlp.DecodeBytes(msg.Msg, &announceData)
+	// Check if the sender is within the registered/elected valset
+	regAndActiveVals, err := sb.retrieveActiveAndRegisteredValidators()
 	if err != nil {
-		logger.Warn("Error in decoding received Istanbul Announce message content", "err", err, "IstanbulMsg", msg.String())
 		return err
 	}
 
-	logger = logger.New("msgAddress", msg.Address, "msg_timestamp", announceData.Timestamp)
-
-	if currentEntryTimestamp, err := sb.valEnodeTable.GetTimestampFromAddress(msg.Address); err == nil {
-		if announceData.Timestamp < currentEntryTimestamp {
-			logger.Trace("Received an old announce message", "currentEntryTimestamp", currentEntryTimestamp)
-			return errOldAnnounceMessage
-		}
-	} else if err != leveldb.ErrNotFound {
-		logger.Warn("Error when retrieving timestamp for entry in the ValEnodeTable", "err", err)
+	if !regAndActiveVals[msg.Address] {
```

Comment:
We chatted about having a signature check. Is this where that should be?

Reply:
Nice catch!  That check is just currently checking if the signer of the message is in the current valset and is not the original message sender.

Reply:
Actually, the code already does check for that here:  https://github.com/celo-org/celo-blockchain/blob/master/consensus/istanbul/types.go#L375.